zipkin-server/src/main/resources/zipkin-server-shared.yml

```diff
@@ -164,6 +164,7 @@ zipkin:
       health-check:
         enabled: ${ES_HEALTH_CHECK_ENABLED:true}
         interval: ${ES_HEALTH_CHECK_INTERVAL:3s}
+      priority: ${ES_TEMPLATE_PRIORITY:}
```

Comment:
sorry I didn't notice this earlier: rename to template-priority

add a test in ZipkinElasticsearchStorageConfigurationTest to ensure that `zipkin.storage.elasticsearch.template-priority:` coerces to null not zero

Reply:
Would you prefer ES_TEMPLATE_PRIORITY= <--- (aka null)  to default to 0, instead of using legacy template? With the type set to nullable Integer, if the ES_TEMPLATE_PRIORITY is not integer (ES_TEMPLATE_PRIORITY=asfsd, for instance), Spring will throw error and fail start up. Andif ES_TEMPLATE_PRIORITY= (null), then legacy template will be used.